x/ethbridge/types/msgs.go

```diff
@@ -49,19 +38,19 @@ func (msg MsgLock) Type() string { return "lock" }
 
 // ValidateBasic runs stateless checks on the message
 func (msg MsgLock) ValidateBasic() error {
-	if strconv.Itoa(msg.EthereumChainID) == "" {
-		return sdkerrors.Wrapf(ErrInvalidEthereumChainID, "%d", msg.EthereumChainID)
+	if msg.EthereumChainId == 0 {
+		return sdkerrors.Wrapf(ErrInvalidEthereumChainID, "%d", msg.EthereumChainId)
 	}
 
-	if msg.CosmosSender.Empty() {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.CosmosSender.String())
+	if msg.CosmosSender == "" {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.CosmosSender)
 	}
 
-	if msg.EthereumReceiver.String() == "" {
+	if msg.EthereumReceiver == "" {
```

Comment:
Have we changed ethereum receiver to not be of type ethereum address? If so, we should change it back. Addresses of cosmos or ethereum type should not be treated as strings in the structures we store them in.